src/arch-x86_64-syscalls.c.perf

```diff
@@ -0,0 +1,565 @@
+sync_file_range2, __PNR_sync_file_range2, 408
+syncfs, 306, 409
+syscall, __PNR_syscall, 410
+sys_debug_setcontext, __PNR_sys_debug_setcontext, 411
+sysfs, 139, 412
+sysinfo, 99, 413
+syslog, 103, 414
+sysmips, __PNR_sysmips, 415
+tee, 276, 416
+tgkill, 234, 417
+time, 201, 418
+timer_create, 222, 419
+timer_delete, 226, 420
+timer_getoverrun, 225, 421
+timer_gettime, 224, 422
+timer_gettime64, __PNR_timer_gettime64, 423
+timer_settime, 223, 424
+timer_settime64, __PNR_timer_settime64, 425
+timerfd, __PNR_timerfd, 426
+timerfd_create, 283, 427
+timerfd_gettime, 287, 428
+timerfd_gettime64, __PNR_timerfd_gettime64, 429
+timerfd_settime, 286, 430
+timerfd_settime64, __PNR_timerfd_settime64, 431
+times, 100, 432
+tkill, 200, 433
+truncate, 76, 434
+truncate64, __PNR_truncate64, 435
+tuxcall, 184, 436
+ugetrlimit, __PNR_ugetrlimit, 437
+ulimit, __PNR_ulimit, 438
+umask, 95, 439
+umount, __PNR_umount, 440
+umount2, 166, 441
+uname, 63, 442
+unlink, 87, 443
+unlinkat, 263, 444
+unshare, 272, 445
+uselib, 134, 446
+userfaultfd, 323, 447
+usr26, __PNR_usr26, 448
+usr32, __PNR_usr32, 449
+ustat, 136, 450
+utime, 132, 451
+utimensat, 280, 452
+utimensat_time64, __PNR_utimensat_time64, 453
+utimes, 235, 454
+vfork, 58, 455
+vhangup, 153, 456
+vm86, __PNR_vm86, 457
+vm86old, __PNR_vm86old, 458
+vmsplice, 278, 459
+vserver, 236, 460
+wait4, 61, 461
+waitid, 247, 462
+waitpid, __PNR_waitpid, 463
+write, 1, 464
+writev, 20, 465
+%%
+/**
+ * Resolve a syscall name to a number
+ * @param name the syscall name
+ *
+ * Resolve the given syscall name to the syscall number using the syscall table.
+ * Returns the syscall number on success, including negative pseudo syscall
+ * numbers; returns __NR_SCMP_ERROR on failure.
+ *
+ */
+int x86_64_syscall_resolve_name(const char *name)
+{
+	const struct arch_syscall_def_internal *s;
+
+	s = in_word_set(name, strlen(name));
+	if (s == NULL)
+		return __NR_SCMP_ERROR;
+
+	return s->num;
+}
+
+/**
+ * Resolve a syscall number to a name
+ * @param num the syscall number
+ *
+ * Resolve the given syscall number to the syscall name using the syscall table.
+ * Returns a pointer to the syscall name string on success, including pseudo
+ * syscall names; returns NULL on failure.
+ *
+ */
+const char *x86_64_syscall_resolve_num(int num)
+{
+	unsigned int iter;
+
+	for (iter = 0; iter < sizeof(wordlist)/sizeof(wordlist[0]); iter++) {
+		if (wordlist[iter].num == num)
+			return (stringpool + wordlist[iter].name);
+	}
+
+	return NULL;
+}
+
+/**
+ * Iterate through the syscall table and return the syscall mapping
+ * @param spot the offset into the syscall table
+ *
+ * Return the syscall mapping at position @spot or NULL on failure.  This
+ * function should only ever be used internally by libseccomp.
+ *
+ */
+const struct arch_syscall_def *x86_64_syscall_iterate(unsigned int spot)
+{
+	unsigned int iter;
+        /* It is thread-unsafe.  It should be used only for testing.  */
+	static struct arch_syscall_def ret;
```

Comment:
I've used the static trick only because this method is used only for the tests:

```
 * Return the syscall mapping at position @spot or NULL on failure.  This
 * function should only ever be used internally by libseccomp.
```

Otherwise it would be a breaking change